### PR TITLE
Fix set mcrud.enforce=1 issues

### PR DESCRIPTION
**Pull request by wyrzyk**

### CODE REVIEW

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java [hunk 1]
```diff
@@ -1754,4 +1754,18 @@ public void shouldPropagateLastWrite() throws SQLException {
...
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().ignoreSupplier(true).build()
+        ).build();
+
+        connection.createStatement().execute("set mcrud.enforce=1");
+
+        assertThat(connectionProvider.getProvidedConnectionTypes())
+            .containsOnly(MAIN);
```
[the remark this answers is not in the document]
Comment (wyrzyk) [reply src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1769]: It shouldn't. I'm fixing the test.

[the remark this answers is not in the document]
Comment (wyrzyk) [reply src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1769]: So the issue is covered with `shouldKeepStatementTimeoutWhenSwitchingConnections`. 🤔

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java [hunk 2]
```diff
@@ -1754,4 +1754,18 @@ public void shouldPropagateLastWrite() throws SQLException {
...
+
+    @Test
+    public void shouldSetEnforceOnMain() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().ignoreSupplier(true).build()
+        ).build();
+
+        connection.createStatement().execute("set mcrud.enforce=1");
```
Comment (afaruga-atlassian) [inline src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:1766]: `SET mcrud.enforce=1` would work with replica?

**Review by afaruga-atlassian — APPROVED** [review]
Repro LGTM, I guess you want to fix it too. Don't know if withing current PR or next one.

**Pull request closed without merging**